### PR TITLE
Add a readme to Beagle Web gRPC

### PR DESCRIPTION
**Pull request by arthurbleilzup**
There was no description on the readme, so now it has the same description as on docs

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by jamessonpenhazup — APPROVED** [review]

File: web/beagle-grpc/README.md [hunk 1]
```diff
@@ -1 +1,148 @@
-# Beagle WEB gRPC
\ No newline at end of file
+# Beagle WEB gRPC
+
+## What is gRPC?
+----------------
+
+It is an open source framework to perform RPC (Remote call procedure) calls, through the HTTP2 protocol in a two-way streaming format, using Protobufs as interfaces between client and server.
```
Comment (igorgiumellizup) [inline web/beagle-grpc/README.md:6]:
```suggestion
It is an open source framework to perform RPC (Remote Call Procedure) calls, through the HTTP2 protocol in a two-way streaming format, using Protobufs as interfaces between client and server.
```

File: web/beagle-grpc/README.md [hunk 2]
```diff
@@ -1 +1,148 @@
...
+
+Protobufs, or Protocol Buffers, are a structured data serialization mechanism created by Google, which aim to keep contracts always valid and consistent regarding the implementations that consume a gRPC service.
```
Comment (igorgiumellizup) [inline web/beagle-grpc/README.md:8]:
```suggestion
Protobufs, or Protocol Buffers, are a structured data serialization mechanism created by Google, which aims to keep contracts always valid and consistent, regarding the implementations that consumes a gRPC service.
```

File: web/beagle-grpc/README.md [hunk 3]
```diff
@@ -1 +1,148 @@
...
+       },
+       components: {},
+       fetchData: usingBeagleGrpcClient({ proxyAddress: 'http://localhost:8081' })
+     })
+     export class Beagle {}
+     ```
+3. Start your proxy running the following command on the root of the project: `beagle-web-grpc start-proxy` or `beagle-web-grpc spx` or `beagle-web-grpc start-proxy --mode your-mode`.
+4. Use the Beagle Web as usually.
+
+> **Observations**
```
Comment (igorgiumellizup) [inline web/beagle-grpc/README.md:77]:
```suggestion
> **Note:**
```

File: web/beagle-grpc/README.md [hunk 4]
```diff
@@ -1 +1,148 @@
...
+> Make sure that your BFF Application is running.
+
+## CLI
+------
+
+As gRPC is not compatible with the current structure of the Web, to support the technology, it is necessary to use a Proxy to establish the communication between the REST application and the gRPC Backend application, so we created a CLI (command-line interface), **beagle-web-grpc**, to facilitate the use of the proxy and provide all the necessary settings.
```
Comment (igorgiumellizup) [inline web/beagle-grpc/README.md:83]:
```suggestion
As gRPC is not compatible with the current structure of the Web, to support the technology, it is necessary to use a proxy to establish the communication between the REST application and the gRPC backend application, so we created a CLI (command-line interface), **beagle-web-grpc**, to facilitate the use of the proxy and provide all the necessary settings.
```

File: web/beagle-grpc/README.md [hunk 5]
```diff
@@ -1 +1,148 @@
...
+
+#### Available commands for **beagle-web-grpc**
+
+| Command       | Alias      | Options     | Description                                                                                       |
+| ------------- | ---------- | ----------- | ------------------------------------------------------------------------------------------------- |
+| `init`        | `i`        |             | Creates the configuration file which define the settings to be used by the Beagle Web gRPC Proxy. |
```
Comment (igorgiumellizup) [inline web/beagle-grpc/README.md:89]:
```suggestion
| `init`        | `i`        |             | Creates the configuration file which defines the settings to be used by the Beagle Web gRPC Proxy. |
```

File: web/beagle-grpc/README.md [hunk 6]
```diff
@@ -1 +1,148 @@
...
+  ]
+}
+```
+
+##### Attributes of the configuration file *./beagle-grpc.config.json*
+
+| Attribute               | Description                                                                   | 
+| ----------------------- | ----------------------------------------------------------------------------- |
+| `mode`                  | Defines the mode name for the settings to be used when running the gRPC Proxy |
+| `grpcBackendAddress`    | Defines the address of the backend server that communicates using gRPC. Remembering that all requests that match this route will be redirected to the Proxy |
```
Comment (igorgiumellizup) [inline web/beagle-grpc/README.md:122]:
```suggestion
| `grpcBackendAddress`    | Defines the address of the backend server that communicates using gRPC. Keep it mind that all requests that match this route will be redirected to the proxy |
```

File: web/beagle-grpc/README.md [hunk 7]
```diff
@@ -1 +1,148 @@
...
+| `tlsCertificatePath`    | Set path to TLS Certificate location                                          |
```
Comment (igorgiumellizup) [inline web/beagle-grpc/README.md:123]:
```suggestion
| `tlsCertificatePath`    | Sets the path to the TLS Certificate location                                          |
```

File: web/beagle-grpc/README.md [hunk 8]
```diff
@@ -1 +1,148 @@
...
+| `tlsKeyPath`            | Set path to TLS Key location                                                  |
```
Comment (igorgiumellizup) [inline web/beagle-grpc/README.md:124]:
```suggestion
| `tlsKeyPath`            | Sets the path to the TLS Key location                                                  |
```

File: web/beagle-grpc/README.md [hunk 9]
```diff
@@ -1 +1,148 @@
...
+| `runProxyOnPort`        | Defines the Port on which the gRPC Proxy will run                             |
+
+## Client
+----------
+
+The gRPC Client for Beagle Web is what redirects grpc requests to the proxy, the client is used through `BeagleService` in React and the `Beagle` module in Angular, using the `fetchData` property, which Beagle allows you to create custom HTTP requests, so the use goes like this:
```
Comment (igorgiumellizup) [inline web/beagle-grpc/README.md:130]:
```suggestion
The gRPC Client for Beagle Web redirects gRPC requests to the proxy. The client is used through `BeagleService` in React and the `Beagle` module in Angular, using the `fetchData` property. Beagle allows you to create custom HTTP requests, so the use goes like this: 
```

File: web/beagle-grpc/README.md [hunk 10]
```diff
@@ -1 +1,148 @@
...
+  // { ... } - other attributes
+  fetchData: usingBeagleGrpcClient({ proxyAddress: 'http://localhost:8081' })
+}
+```
+
+The gRPC client has the following initialization options (`BeagleGrpcClientOptions`):
+
+| Attribute             | Required?       | Default Value   | Description                                                                           |
+| --------------------- | --------------- | --------------- | ------------------------------------------------------------------------------------- |
+| `proxyAddress`        | Yes             |                 | Address where the Proxy will be running for the client to communicate with the proxy  |
```
Comment (igorgiumellizup) [inline web/beagle-grpc/README.md:146]:
```suggestion
| `proxyAddress`        | Yes             |                 | Address where the proxy will be running for the client to communicate with the proxy  |
```

**Pull request closed without merging**